bin/tests/integration/server_harness.rs

```diff
@@ -118,152 +119,105 @@ where
             "--zonedir={server_path}/tests/test-data/test_configs"
         ))
         .arg(format!("--port={}", 0));
-    #[cfg(feature = "__tls")]
-    command.arg(format!("--tls-port={}", 0));
-    #[cfg(feature = "__https")]
-    command.arg(format!("--https-port={}", 0));
-    #[cfg(feature = "__quic")]
-    command.arg(format!("--quic-port={}", 0));
-    #[cfg(feature = "prometheus-metrics")]
-    command.arg(format!("--prometheus-listen-address=127.0.0.1:{}", 0));
-
-    println!("named cli options: {command:#?}");
-
-    let mut named = command.spawn().expect("failed to start named");
-
-    println!("server starting");
-
-    let mut named_out = BufReader::new(named.stdout.take().expect("no stdout"));
-
-    // forced thread killer
-    let named = Arc::new(Mutex::new(named));
-    let named_killer = Arc::clone(&named);
-    let succeeded = Arc::new(atomic::AtomicBool::new(false));
-    let succeeded_clone = succeeded.clone();
-    let killer_join = thread::Builder::new()
-        .name("thread_killer".to_string())
-        .spawn(move || {
-            let succeeded = succeeded_clone;
-
-            let kill_named = || {
-                info!("killing named");
-
-                let mut named = named_killer.lock().unwrap();
-                if let Err(error) = named.kill() {
-                    warn!(?error, "warning: failed to kill named");
-                    return;
-                }
-                if let Err(error) = named.wait() {
-                    warn!(?error, "warning: failed to wait for named");
-                }
-            };
+        #[cfg(feature = "__tls")]
+        command.arg(format!("--tls-port={}", 0));
+        #[cfg(feature = "__https")]
+        command.arg(format!("--https-port={}", 0));
+        #[cfg(feature = "__quic")]
+        command.arg(format!("--quic-port={}", 0));
+        #[cfg(feature = "prometheus-metrics")]
+        command.arg(format!("--prometheus-listen-address=127.0.0.1:{}", 0));
 
-            for _ in 0..30 {
-                thread::sleep(Duration::from_secs(1));
-                if succeeded.load(atomic::Ordering::Relaxed) {
-                    kill_named();
-                    return;
-                }
-            }
+        println!("named cli options: {command:#?}");
+
+        let mut named = command.spawn().expect("failed to start named");
 
-            kill_named();
+        println!("server starting");
 
-            println!("Thread Killer has been awoken, killing process");
-            std::process::exit(-1);
-        })
-        .expect("could not start thread killer");
+        let mut stdout = BufReader::new(named.stdout.take().expect("no stdout"));
 
-    // These will be collected from the server startup'
-    let mut socket_ports = SocketPorts::default();
+        // These will be collected from the server startup'
```

Comment:
nit: pre-existing typo, a trailing `'` Maybe should read "These will be collected from the server's startup output" ?